test/fakeintake/aggregator/apmStatsAggregator.go

```diff
@@ -0,0 +1,66 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package aggregator
+
+import (
+	"time"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/test/fakeintake/api"
+
+	"github.com/tinylib/msgp/msgp"
+)
+
+// APMStatsPayload is a payload type for APM stats
+// It implements PayloadItem
+type APMStatsPayload struct {
+	*pb.StatsPayload
+	collectedTime time.Time
+}
```

Comment:
I don’t know the APM payloads well enough. So maybe what I’m about to suggest doesn’t apply for this usecase.
But, would it make sense to have the fake intake client unbundle the messages that have been grouped by the agent in a single message ?

I mean:
```suggestion
type APMStatsPayload struct {
	*pb.ClientStatsPayload
	collectedTime time.Time
}
```

and then… (see next comment)

Reply:
It's indeed an interesting idea but in that case we would miss top level fields like `AgentHostname` `AgentEnv` `AgentVersion` `ClientComputed` and `SplitPayload` which I believe we'll need in the tests later. For instance `ClientComputed` is crucial to know whether the payload was computed by the tracer or the agent and we'll need to verify that field in the tests.